app/src/main/java/com/example/henzoshimada/feeltrip/MoodAdapter.java

```diff
@@ -78,16 +83,27 @@ public View getView(int position, View convertView, ViewGroup parent) {
         lastPosition = position;
 */
         Log.d("imageTag","new array list");
+        assert mood != null;
         viewHolder.userName.setText(mood.getUsername());
         viewHolder.date.setText(mood.getDate().toString());
         viewHolder.description.setText(Html.fromHtml(mood.getDescription())); //TODO: This is depreciated, maybe replace?
-        viewHolder.append.setText(" -Feeling " + mood.getEmotionalState());
-        viewHolder.emoji.setImageBitmap(null); //TODO: pass the emoji into here
+        viewHolder.append.setText(" - Feeling " + mood.getEmotionalState());
         viewHolder.socialSituation.setText(mood.getSocialSit());
 
+        //#######################################################
+        Log.i("Mytag","Emoooooo: "+mood.getEmotionalState());
+        int emojiID = getContext().getApplicationContext().getResources().getIdentifier("emoji" + String.valueOf(mood.getEmoji()),"drawable",getContext().getApplicationContext().getPackageName());
+        if(emojiID != 0) {
+            viewHolder.emojiImage.setImageResource(emojiID);
+        }
```

Comment:
It was working fine, but it was statically declared, it could ONLY handle those eight emotions, this is simply part of the EASY loading process.